addons/rating/models/mail_thread.py

```diff
@@ -1,12 +1,178 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models, _
+import datetime
+import markupsafe
+
+from odoo import api, fields, models, _, tools
 
 
 class MailThread(models.AbstractModel):
     _inherit = 'mail.thread'
 
+    rating_ids = fields.One2many('rating.rating', 'res_id', string='Rating', groups='base.group_user',
+                                 domain=lambda self: [('res_model', '=', self._name)], auto_join=True)
+
+    def unlink(self):
+        """ When removing a record, its rating should be deleted too. """
+        record_ids = self.ids
+        result = super(MailThread, self).unlink()
+        self.env['rating.rating'].sudo().search([('res_model', '=', self._name), ('res_id', 'in', record_ids)]).unlink()
+        return result
+
+    def _message_create(self, values_list):
```

Comment:
Linting/Suggestion: We could make sections in the file using "comments", just to make the file easy to browse, like
```
    # MAIL OVERRIDES
    # --------------------------------------------------
    ....

    # RATING SUPPORT
    # --------------------------------------------------
    ...
```

Reply:
Yes good idea as we just populated the file from the mixin and it was almost empty.